### PR TITLE
feat(custom-field): expose structured custom-field values (people gid, enum/multi_enum options, date ISO) in task custom-field list

### LINKED ISSUE #74: feat(custom-field): expose people_value gid in `task custom-field list` to enable @mentions
## Summary

`asana task custom-field list <task-gid>` cannot surface the **user gid** of a `people`-type
custom field. It only exposes `display_value` (a name string). As a result, callers can read
*who* a people field points to by name, but cannot build a real Asana **@-mention** in a
comment, because a mention requires the user's gid.

## Current behavior

`task custom-field list` requests a fixed, minimal set of `opt_fields` and maps only the
display value:

- `src/commands/custom-field.ts:12`
  ```ts
  opt_fields: 'custom_fields.name,custom_fields.resource_subtype,custom_fields.display_value',
  ```
- `src/commands/custom-field.ts:77`
  ```ts
  const fields = (task.custom_fields || []).map((field: any) => ({
    gid: field.gid,
    name: field.name,
    type: field.resource_subtype ?? 'unknown',
    value: field.display_value ?? '',   // people field → names only, no user gid
  }))
  ```

So for a `people` custom field the output carries names but never the assigned users' gids.

(For context, `asana task get` does not return custom fields at all, so `custom-field list`
is the only read path.)

## Why it matters

The comment command already supports rich text:

- `src/commands/task-comment.ts:38`
  ```ts
  .option('--html', 'Treat text as rich text (Asana HTML, wrapped in <body> automatically)')
  ```

Asana renders an @-mention from `html_text` containing `<a data-asana-gid="USER_GID"/>`. The
only missing piece is the user gid. With it, an integration could read a "reporter"/"보고자"
people field and post a comment that actually @-mentions and notifies that person.

Concrete use case: a finalize/automation step that posts a resolution note to the task and
@-mentions the reporter stored in a people custom field.

## API supports this

This is purely a CLI gap — the Asana API already exposes the data. A `people` custom field
returns a `people_value` array of compact user objects, each with a `gid`
(https://developers.asana.com/reference/custom-fields). `display_value` is documented as a
string representation for integrations that don't need the underlying type.

## Proposed change

1. Add the people value (with gid + name) to the `opt_fields` in `TASK_CUSTOM_FIELD_FIELDS`:
   ```ts
   opt_fields: 'custom_fields.name,custom_fields.resource_subtype,custom_fields.display_value,custom_fields.people_value.gid,custom_fields.people_value.name',
   ```
2. Include the structured people value in the `list` output for `people`-type fields, e.g. a
   `people: [{ gid, name }]` array alongside the existing `value` (display string), so JSON
   consumers (`--format json`) can read gids without breaking the current `value` shape.

This is additive and backward compatible: `value`/`display_value` stay as-is.

## Acceptance criteria

- [ ] `asana task custom-field list <gid> --format json` returns, for a `people`-type field,
      the assigned users' `gid`s (not just names).
- [ ] Existing fields (`gid`, `name`, `type`, `value`) are unchanged for non-people types.
- [ ] A short note/example in the skill docs shows how to combine the people gid with
      `task comment add --html` to @-mention a person.

## Environment

- Repo: `pleaseai/asana` @ `main`
- Verified against source files referenced above.

### PR DESCRIPTION
**Pull request by amondnet**
## Summary

`task custom-field list` only exposed `display_value` (a display string) for every
custom field type, which hid machine-readable data callers need. The primary gap
(#74) was that **people** fields gave names but never the user **gid**, so callers
could not build a real Asana @-mention. While extending the read path, the same
`display_value`-only gap for `enum`/`multi_enum`/`date` fields is closed too.

## Changes

For each type whose `display_value` hides machine-readable data, a structured key
is added to the `list` output (and the matching `opt_fields` are requested):

| type | new key | shape |
|------|---------|-------|
| `people` | `people` | `[{ gid, name }]` — user gids for @-mentions |
| `enum` | `enum_option` | `{ gid, name }` or `null` if unset |
| `multi_enum` | `enum_options` | `[{ gid, name }]` |
| `date` | `date` | `{ date, date_time }` raw ISO, or `null` if unset |

- Routes every type through a unit-tested `mapTaskCustomField` helper.
- **Fully additive / backward compatible**: `gid`, `name`, `type`, `value`
  (= `display_value`) are unchanged for every type, and each structured key
  appears only for its own field type.
- Updated `skills/asana-cli/references/commands.md` documenting all structured
  keys plus a worked example combining the people gid with
  `task comment add --html` to @-mention a person.

## Test Plan

- [x] 13 `mapTaskCustomField` unit tests covering people / enum / multi_enum / date
      mapping, unset values (`null`/`[]`), and non-matching-type passthrough.
- [x] Full test suite: 449 pass / 0 fail.
- [x] ESLint clean.

## Related Issues

Closes #74

## Summary by CodeRabbit

* **New Features**
  * Custom-field listings now provide richer, structured summaries, including people selections, single/multi-enum options, and date details.
  * Documentation added for mentioning a person from a people-type custom field using machine-readable JSON fields.

* **Bug Fixes**
  * Improved handling of missing or empty custom-field values with consistent defaults and correctly nullable structured outputs.

* **Documentation**
  * Expanded “task custom-field” reference with an end-to-end example for HTML mentions.

* **Tests**
  * Added/expanded unit coverage for mapping behavior across multiple custom-field types and edge cases.

### CODE REVIEW

File: src/lib/custom-fields.ts [hunk 1]
```diff
@@ -21,6 +21,50 @@ export interface CustomFieldDefinition {
...
+
+  if (type === 'people') {
+    summary.people = (field.people_value ?? []).map((user: any) => ({
+      gid: user.gid,
+      name: user.name,
+    }))
+  }
+
+  return summary
+}
```
[the remark this answers is not in the document]
Comment (amondnet) [reply src/lib/custom-fields.ts:110]: @gemini-code-assist 검토했는데 이번엔 적용하지 않겠습니다. 🙏

제안의 핵심인 `if (!field)` → 빈 summary(`{gid:'', type:'unknown'}`) 반환은 잘못된 입력을 조용히 삼켜 버그를 가리는 **silent-failure** 패턴이라, 차라리 throw 되는 편이 디버깅에 낫습니다. 그리고 `mapTaskCustomField`의 유일한 호출부(`src/commands/custom-field.ts:89`)는 `(task.custom_fields || []).map(mapTaskCustomField)` 로 Asana API가 돌려준 필드 객체 배열만 넘기므로 `field`가 null/undefined가 되는 경로는 없습니다.

`people_value` user 필터(`user?.gid`)도 `opt_fields`에서 `custom_fields.people_value.gid`를 명시적으로 요청하기 때문에 gid 없는 user가 들어오지 않습니다.

두 방어 코드 모두 도달 불가능한 경로(dead code)를 만들고 STANDARDS.md의 'unjustified defensive code 지양' 원칙과 충돌해서, 입력 계약을 호출부에서 보장하는 현재 방식을 유지하겠습니다.

File: src/lib/custom-fields.ts [hunk 2]
```diff
@@ -21,6 +21,92 @@ export interface CustomFieldDefinition {
...
+ * machine-readable data, the structured value is surfaced as an extra key so
+ * JSON consumers can read it:
+ *
+ * - `people` → `people`: assigned users with their `gid`s (needed for @-mentions)
+ * - `enum` → `enum_option`: the selected option `{gid, name}`, or `null`
+ * - `multi_enum` → `enum_options`: the selected options `[{gid, name}]`
+ * - `date` → `date`: the raw `{date, date_time}` ISO value, or `null`
+ */
+export function mapTaskCustomField(field: any): TaskCustomFieldSummary {
+  const type = field.resource_subtype ?? 'unknown'
```
[the remark this answers is not in the document]
Comment (amondnet) [reply src/lib/custom-fields.ts]: @coderabbitai 적용했습니다. ✅

`mapTaskCustomField`의 타입 선택을 `field.resource_subtype ?? field.type ?? 'unknown'` 으로 바꿔서 `coerceCustomFieldValue`(같은 파일 32번 줄)와 동일하게 legacy `type` shape도 올바른 타입으로 매핑되도록 했습니다. legacy fallback 회귀 테스트도 추가했고 전체 스위트 450 pass / 0 fail, lint clean 입니다. (commit ecb04a9)
